src/dfx/src/commands/start.rs

```diff
@@ -209,6 +218,8 @@ pub fn exec(
         empty_writable_path(local_server_descriptor.icx_proxy_pid_path())?;
     let webserver_port_path = empty_writable_path(local_server_descriptor.webserver_port_path())?;
 
+    let previous_config_path = network_temp_dir.join("replica-effective-config.json");
```

Comment:
DRY + "one of these things is not like the other"
```suggestion
    let previous_config_path = local_server_descriptor.replica_effective_config_path();
```